homeassistant/components/opentherm_gw/__init__.py

```diff
@@ -115,14 +116,18 @@
         DATA_GW_VARS: pyotgw.vars,
         DATA_LATEST_STATUS: {}
     }
-    hass.async_create_task(connect_and_subscribe(
-        hass, conf[CONF_DEVICE], gateway))
     hass.async_create_task(register_services(hass, gateway))
     hass.async_create_task(async_load_platform(
         hass, 'climate', DOMAIN, conf.get(CONF_CLIMATE), config))
     if monitored_vars:
         hass.async_create_task(setup_monitored_vars(
             hass, config, monitored_vars))
+
+    def schedule_connect(event):
+        """Schedule the connect_and_subscribe coroutine."""
+        hass.async_create_task(
```

Comment:
Instead of waiting until home assistant start event, we can schedule the task directly on the loop with `hass.loop.create_task`. Then the task won't be tracked by home assistant and set up phase can finish.

Reply:
I like your suggested solution better than the one in this PR. Expect a new PR soon.